controller/node_controller.go

```diff
@@ -1607,6 +1613,46 @@ func (nc *NodeController) deleteDisk(diskType longhorn.DiskType, diskName, diskU
 	return nil
 }
 
+func (nc *NodeController) cleanupDisksBeforeNodeDeletion(node *longhorn.Node) error {
+	nc.logger.Info("Cleaning up disks before node deletion")
+
+	isV2DataEngine, err := nc.ds.GetSettingAsBool(types.SettingNameV2DataEngine)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get v2 data engine setting")
+	}
+
+	if !isV2DataEngine {
+		nc.logger.Info("Skipping disk cleanup as v2 data engine is not enabled")
+		return nil
+	}
+
+	for diskName, diskInfo := range node.Spec.Disks {
+		nc.logger.Infof("Cleaning up disk %s", diskName)
+		// Skip non-SPDK disks
+		if diskInfo.Type != longhorn.DiskTypeBlock {
+			continue
+		}
+
+		// Retrieve disk status
+		diskStatus, ok := node.Status.DiskStatus[diskName]
+		if !ok {
+			nc.logger.Infof("Disk %s exists in spec but is missing from status. Skipping SPDK cleanup.", diskName)
+			continue
+		}
+
+		diskInstanceName := diskStatus.DiskName
```

Comment:
Don't we expect the value should be identical? If yes, why not just use `diskName` or `diskStatus.DiskName` directly w/o this fallback design?